api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -833,6 +833,14 @@ type AWSNodePoolPlatform struct {
 	// +optional
 	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
 
+	// Placement specifies the placement options for the EC2 instances.
+	//
+	// +optional
+	Placement *PlacementOptions `json:"placement,omitempty"`
+}
+
+// PlacementOptions specifies the placement options for the EC2 instances.
+type PlacementOptions struct {
```

Comment:
I wanted to follow the AWS SDK naming convention which uses `Placement`. 
> in fact capa already have PlacementGroupPartition and PlacementGroupName

right, those fields are also part of the same placement input in the SDK, I don't know why they are not grouped in CAPA, but If we ever expose them to our API, we can add them to the same placement struct, which is why I think tenacnyOptions might not be a good naming in the future.

happy to change it if you disagree.

Reply:
ok, that makes sense to me

Reply:
A "placement group" is a concept within AWS, so I wonder if the placement group related options are separate because otherwise they would have to be `placement.placementGroup{Name|Partition}`?

Reply:
![image](https://github.com/user-attachments/assets/c5699060-4715-4a9f-a342-7f0f7a68618a)

from: https://github.com/aws/aws-sdk-go/blob/main/service/ec2/api.go

Reply:
"placement group" is also part of the same input in the SDK convention like along `tenancy`, so it would be `placement.group{Name|Partition|Id}`